### PR TITLE
feat: removed usage of number when swapping

### PR DESCRIPTION
**Pull request by glo12345**

### CODE REVIEW

File: web/app.fluidity.money/app/components/Fluidify/FluidifyForm.tsx [hunk 1]
```diff
@@ -26,30 +28,57 @@ export const FluidifyForm = ({
...
 
-  const swapAmount: number = snapToValidValue(swapInput);
+    const maxMintable = assetToken.userMintLimit.sub(
+      assetToken.userMintedAmt || new BN(0)
+    );
+
+    return BN.min(maxUserPayable, maxMintable);
+  };
+
+  const swapAmount: BN = snapToValidValue(swapInput);
```
Comment (eli-d) [inline web/app.fluidity.money/app/components/Fluidify/FluidifyForm.tsx:68]: maybe could be a useEffect?

File: web/app.fluidity.money/app/util/chainUtils/tokens.ts [hunk 2]
```diff
@@ -65,9 +66,31 @@ const fluidAssetOf = (tokens: Token[], assetToken: Token): Token | undefined =>
...
+const getUsdFromTokenAmount = (amount: BN, { decimals }: Token) => {
+  return amount.div(new BN(10).pow(new BN(decimals - 2))).toNumber() / 100;
+};
+
+const getTokenAmountFromUsd = (usd: BN, { decimals }: Token) =>
+  usd.mul(new BN(10).pow(new BN(decimals)));
+
+// Format BN with decimals
+const addDecimalToBn = (amount: BN, decimals: number) => {
+  const whole = amount.toString().slice(0, -decimals) || "0";
```
Comment (eli-d) [inline web/app.fluidity.money/app/util/chainUtils/tokens.ts:78]: i don't think this pads values correctly. e.g. with 6 decimals 10, 100, 1000 all give the same output (0.1).

File: web/app.fluidity.money/app/components/Fluidify/FluidifyForm.tsx (hunk 1 above)
Comment (glo12345) [reply web/app.fluidity.money/app/components/Fluidify/FluidifyForm.tsx:68]: there's only one state in this comp, so unnecessary

File: web/app.fluidity.money/app/util/chainUtils/tokens.ts (hunk 2 above)
Comment (glo12345) [reply web/app.fluidity.money/app/util/chainUtils/tokens.ts:78]: good catch - fixed

**Review by eli-d — APPROVED** [review]

**Pull request merged**